### PR TITLE
Added mount, umount and rename

### PR DESCRIPTION
**Pull request by apatil**
Added wrappers for zfs mount, zfs umount and zfs rename.

[Review on Reviewable](https://reviewable.io/reviews/mistifyio/go-zfs/33)

### CODE REVIEW

File: zfs.go [hunk 1]
```diff
@@ -172,6 +172,44 @@ func (d *Dataset) Clone(dest string, properties map[string]string) (*Dataset, er
 	return GetDataset(dest)
 }
 
+func (d *Dataset) Umount(force bool) (*Dataset, error) {
```
Comment (mmlb) [inline zfs.go]: Missing a comment

File: zfs.go [hunk 2]
```diff
@@ -172,6 +172,44 @@ func (d *Dataset) Clone(dest string, properties map[string]string) (*Dataset, er
...
+	}
+	args = append(args, d.Name)
+	_, err := zfs(args...)
+	if err != nil {
+		return nil, err
+	}
+	return GetDataset(d.Name)
+}
+
+func (d *Dataset) Mount(overlay bool, options []string) (*Dataset, error) {
```
Comment (mmlb) [inline zfs.go]: ditto

File: zfs.go [hunk 3]
```diff
@@ -172,6 +172,44 @@ func (d *Dataset) Clone(dest string, properties map[string]string) (*Dataset, er
 	return GetDataset(dest)
 }
 
+func (d *Dataset) Umount(force bool) (*Dataset, error) {
+	if d.Type == DatasetSnapshot {
+		return nil, errors.New("cannot umount snapshots")
+	}
+	args := make([]string, 1, 3)
+	args[0] = "umount"
```
Comment (trisk) [inline zfs.go]: The actual documented subcommand is `unmount`, with `umount` as syntactic sugar.

**Comment by mmlb** [comment]
merged in 16b13ca

**Pull request closed without merging**